### ISSUE TITLE
The way to assert that matched element is NOT present in the list

### ISSUE THREAD

**Issue by viktor-klymenko** [serenity-bdd/serenity-core#453]
Hi guys,
I tried to perform verification that particular element is not present in the List of Map and seems to me that there are no straightforward way to do this using BeanMatchers and static method `shouldMatch(List<T> items, BeanMatcher... matchers)`
If I try to do something like this
`shouldMatch(items, the("id", is(not("12"))))`
assert will never fail because there will be at least one item with "id" is not "12" (unless there is only one item and it has id "12")

Is it possible for you to add new static method into the BeanMatcherAsserts class which will allow to verify that list doesn't contain element that matches to given matcher?
I tried to create the following method based on shouldMatch method and it works fine:
```
public static <T> void shouldNotMatch(List<T> items, BeanMatcher... matchers) {
        if (matches(items, matchers)) {
            throw new AssertionError("Found unneeded matching elements for " + join(matchers)
                                     + NEW_LINE
                                     +"Elements where " + join(items));
        }
    }

```
and now I can use it in a "fluent" way:
`shouldNotMatch(items, the("id", is("12")))`

**Comment by wakaleo** [comment]
Sure, that would not be hard - you could even bundle what you have done
with some unit tests and submit a pull request if you want to fast track
things ;-)

___________________________________________________
John Smart | Wakaleo Consulting  |  +44 7398 832273
Making smart teams collaborate better

http://johnfergusonsmart.com  | reachme@johnfergusonsmart.com
<john.smart@wakaleo.com>___________________________________________________


On 15 June 2016 at 9:00:27 AM, viktor-klymenko (notifications@github.com)
wrote:

Hi guys,
I tried to perform verification that particular element is not present in
the List of Map and seems to me that there are no straightforward way to do
this using BeanMatchers and static method shouldMatch(List<T> items,
BeanMatcher... matchers)
If I try to do something like this
shouldMatch(items, the("id", is(not("12"))))
assert will never fail because there will be at least one item with "id" is
not "12" (unless there is only one item and it has id "12")

Is it possible for you to add new static method into the BeanMatcherAsserts
class which will allow to verify that list doesn't contain element that
matches to given matcher?
I tried to create the following method based on shouldMatch method and it
works fine:

public static <T> void shouldNotMatch(List<T> items, BeanMatcher... matchers) {
        if (matches(items, matchers)) {
            throw new AssertionError("Found unneeded matching elements
for " + join(matchers)
                                     + NEW_LINE
                                     +"Elements where " + join(items));
        }
    }


and now I can use it in a "fluent" way:
shouldNotMatch(items, the("id", is("12")))

—
You are receiving this because you are subscribed to this thread.
Reply to this email directly, view it on GitHub
<https://github.com/serenity-bdd/serenity-core/issues/453>, or mute the
thread
<https://github.com/notifications/unsubscribe/AADaslQZlyh90oES9E4nZY-8S3pNsH8Oks5qL7EUgaJpZM4I2F7z>
.

**Comment by wakaleo** [comment]
Released in 1.1.37-rc.5
